src/Symfony/Component/Console/Application.php

```diff
@@ -941,6 +959,17 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             // ignore invalid options/arguments for now, to allow the event listeners to customize the InputDefinition
         }
 
+        if ($this->signalRegistry) {
+            $event = new ConsoleSignalEvent($command, $input, $output);
```

Comment:
Could you add the signal number (SIGINT, etc) to the event ? 
This mean you will have to move this part of the code in the onStopHandler method